### PR TITLE
Add test script for native training.

### PR DESCRIPTION
**Pull request by dellis23**

### CODE REVIEW

**Comment by dellis23** [comment]
Tested locally.  Is there a way I can test this on buildkite before merging?

**Comment by ScottTodd** [comment]
> Tested locally. Is there a way I can test this on buildkite before merging?

I started a build at https://buildkite.com/iree/iree-samples/builds/554. We can also add you to the buildkite project so you can trigger your own builds.

File: samples/native_training/test.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Update submodules in this repo.
+(cd $(git rev-parse --show-toplevel) && git submodule update --init)
+
+# Build the IREE runtime.
+(cd $(git rev-parse --show-toplevel) && cmake -GNinja -B /tmp/iree-build-runtime/ .)
+cmake --build /tmp/iree-build-runtime/ --target iree_runtime_unified
+
+# Build the example.
+cd $(git rev-parse --show-toplevel)/samples/native_training
+make
```
Comment (ScottTodd) [inline samples/native_training/test.sh:42]: https://buildkite.com/iree/iree-samples/builds/554#0184c9f7-5ff0-483f-979c-3091c6e8a49e

```
[2022-11-30T19:43:57Z] + make
--
  | [2022-11-30T19:43:57Z] /usr/bin/clang++-9 main.c -flto -Os -I../../runtime/src/ -L/tmp/iree-build-runtime//build_tools/third_party/flatcc/ -L/tmp/iree-build-runtime//runtime/src/iree/runtime/ -L/tmp/iree-build-runtime//third_party/cpuinfo/ -L/tmp/iree-build-runtime//third_party/cpuinfo/deps/clog -liree_runtime_unified -lflatcc_parsing -lcpuinfo -lclog -ldl -lpthread -o native-training
  | [2022-11-30T19:43:57Z] clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
  | [2022-11-30T19:43:58Z] /usr/bin/ld: error: Optimization level must be between 0 and 3
  | [2022-11-30T19:43:58Z] clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

you can also debug locally under Docker by copying this command (after configuring Docker on your machine): https://github.com/iree-org/iree/blob/693f45abe4e7e5fea8528c59f4d5e983abd04fdd/build_tools/buildkite/samples.yml#L16-L22

**Review by ScottTodd — APPROVED** [review]

File: samples/native_training/test.sh (hunk 1 above)
Comment (ScottTodd) [reply samples/native_training/test.sh:42]: Yep, that build is passing now: https://buildkite.com/iree/iree-samples/builds/556 (ignore the "Build host install", something else is happening there)

**Pull request merged**